### PR TITLE
Register API integration

### PR DESCRIPTION
**Pull request by seromarin**
## Summary
- Completed register application.

## Trello Card
https://trello.com/c/9SXzr1K8/7-registration-api-integration

### CODE REVIEW

File: wolox-books/src/app/interfaces/user.interface.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
   'first_name': string;
   'last_name': string;
   'email': string;
   'password': string;
   'password_confirmation': string;
 }
+
+export {
+  IUser,
+}
```
Comment (mcallegari10) [inline wolox-books/src/app/interfaces/user.interface.ts]: You can accomplish the same and we usually do this:

```ts
export interface IUser {
  ...
}
```

File: wolox-books/src/environments/environment.prod.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export const environment = {
```
Comment (mcallegari10) [inline wolox-books/src/environments/environment.prod.ts:1]: Add the env files to the gitignore please 😄

**Review by mcallegari10 — APPROVED** [review]

File: wolox-books/src/app/services/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export class UserService {
+
+  private readonly API_URI: string = `${environment.api}/users`;
+
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  createUser(newUser: IUser): Observable<IRegisterResponse> {
+    return this.http.post<IRegisterResponse>(this.API_URI, newUser)
```
Comment (idcuadrowolox) [inline wolox-books/src/app/services/user.service.ts]:
```suggestion
    return this.http.post<IRegisterResponse>(this.API_URI, newUser);
```

File: wolox-books/src/app/screens/register/register.component.ts [hunk 4]
```diff
@@ -15,18 +17,28 @@ export class RegisterComponent{
...
       'email': [ null, [ Validators.required, Validators.email ] ],
       'password': [ null, [ Validators.required, Validators.pattern(patternPasswordValidation) ] ],
       'password_confirmation': [ null, [ Validators.required, passwordConfirmation() ] ],
+      'locale': [ 'es' ]
     })
   }
 
   registerUser(user: IUser) {
-    console.log({ user });
+    this.userService.createUser(user).subscribe((respose: IRegisterResponse) => console.log('success', respose))
```
Comment (idcuadrowolox) [inline wolox-books/src/app/screens/register/register.component.ts]:
```suggestion
    this.userService.createUser(user).subscribe((respose: IRegisterResponse) => console.log('success', respose));
```

**Review by idcuadrowolox — APPROVED** [review]

File: wolox-books/src/app/interfaces/user.interface.ts (hunk 1 above)
Comment (seromarin) [reply wolox-books/src/app/interfaces/user.interface.ts]: Fixed

File: wolox-books/src/environments/environment.prod.ts (hunk 2 above)
Comment (seromarin) [reply wolox-books/src/environments/environment.prod.ts:1]: Changed to example env files

**Pull request merged**